benchmark/matmul.cpp

```diff
@@ -231,17 +236,52 @@ MatmulParams getMatmulParams(
   params.double_buffer_options.double_buffer_smem_write = true;
   params.double_buffer_options.double_buffer_smem_read = true;
   params.double_buffer_options.smem_double_buffer_stage = stage_number;
+  params.splitk_factor = splitk_factor;
 
   return params;
 }
 
+// Compute splitk_factor that would fill a block
+int computeAutoSplitKFactor(
+    int M,
+    int N,
+    int tile_M,
+    int tile_N,
+    int num_SMs = 108) {
```

Comment:
Why not just use:
```C++
at::cuda::getCurrentDeviceProperties()->multiProcessorCount
```

Reply:
See note below about adapting to SM count at runtime. It's tough to do so properly when we have to define all the tests at compile time.

Reply:
How about adding a comment, that the default value is based on A100 details?

I see that this is mentioned in a [comment](https://github.com/NVIDIA/Fuser/pull/1002/files#diff-cfe734a1b4887fe9fcd21bde372b512c98b4b5c5d51f0312f88f22fef43ea840R348), but it would be better to have this here, close to the value.